lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js

```diff
@@ -98,38 +108,112 @@ describe('SubmitGradeForm', () => {
     });
   });
 
-  context('when requests are sent', () => {
+  context('when submitting a grade', () => {
+    beforeEach(() => {
```

Comment:
Can you help me understand the purpose of this `beforeEach`, e.g. I don't see how it applies to the spinner test just below; also, in some other tests, I'm seeing manual calls to `fakeSubmitGrade.resolves`. Is this meant to clear a value out or something?

Reply:
It's just a general area to reset it back to a resolving promise. In one of these tests in this block its changed to throw an error and if its not changed back, then it breaks other tests. Not sure if there is a more obvious way to do it?